src/cryptography/x509/base.py

```diff
@@ -104,6 +104,13 @@ def public_key(self) -> _PUBLIC_KEY_TYPES:
         Returns the public key
         """
 
+    @abc.abstractmethod
```

Comment:
As an internal method we don't need to add it to our interface (or document it publicly)

Reply:
Oho! This makes the PR much shorter.
Branch force-pushed (also updated on `main`).